Orange/tests/test_instance.py

```diff
@@ -124,43 +123,32 @@ def test_init_xy_arr(self):
     def test_init_xy_list(self):
         domain = self.create_domain(["x", DiscreteVariable("g", values="MF")],
                                     [DiscreteVariable("y", values="ABC")])
-        lst = [42, "M", "C"]
         vals = np.array([42, 0, 2])
         inst = Instance(domain, vals)
         assert_array_equal(inst._x, vals[:2])
         self.assertEqual(inst._y.shape, (1, ))
         self.assertEqual(inst._y[0], 2)
         self.assertEqual(inst._metas.shape, (0, ))
 
-    def test_init_xym_arr(self):
+    def helper_init_xym_arr(self, vals):
         domain = self.create_domain(["x", DiscreteVariable("g", values="MF")],
                                     [DiscreteVariable("y", values="ABC")],
                                     self.metas)
-        vals = np.array([42, "M", "B", "X", 43, "Foo"], dtype=object)
         inst = Instance(domain, vals)
         self.assertIsInstance(inst, Instance)
         self.assertIs(inst.domain, domain)
-        self.assertEqual(inst._x.shape, (2, ))
-        self.assertEqual(inst._y.shape, (1, ))
-        self.assertEqual(inst._metas.shape, (3, ))
+        self.assertEqual(inst._x.shape, (2,))
+        self.assertEqual(inst._y.shape, (1,))
+        self.assertEqual(inst._metas.shape, (3,))
         assert_array_equal(inst._x, np.array([42, 0]))
         self.assertEqual(inst._y[0], 1)
         assert_array_equal(inst._metas, np.array([0, 43, "Foo"], dtype=object))
 
+    def test_init_xym_arr(self):
+        self.helper_init_xym_arr(np.array([42, "M", "B", "X", 43, "Foo"], dtype=object))
```

Comment:
Instead of a helper method, you could have single `test_init` and then different `args` applied in a for loop, like with most other tests above.